File: python/Airbnbclone_takeaways/TwoSumTarget.py
def SumTwoNumber_Target(arr, target):
    left = 0
    right = len(arr) - 1
    # Two pointers logic
    while left < right:
        currentSum = arr[left] + arr[right]
        if currentSum == target:
            return True
        elif currentSum < target:
             left += 1
        else:
            right -= 1
    return False
        

print(SumTwoNumber_Target([1,2,3,4,5], 5))
print(SumTwoNumber_Target([1,2,3,4,5], 50))

File: python/Airbnbclone_takeaways/TwoSumTarget.js
//This approach is used with a sorted array

function TwoSumTarget(arr, target) {
    let left = 0;
    let right = arr.length - 1;

    while (left < right) {
        let currentSum = arr[left] + arr[right]
        if (currentSum == target) {
            return true;
        } else if (currentSum < target) {
            left++;
        } else {
            right--;
        }
    }
    return false;
}

console.log(TwoSumTarget([1,2,3,4,5], 5));
console.log(TwoSumTarget([1,2,3,4,5], 50));